### PR TITLE
ISDK-2505: Update QuickStarts for 3.0.0

### PR DESCRIPTION
**Pull request by paynerc**
This PR updates the Swift Quickstarts with the 3.0.0 Swift API changes and adds support for selecting signaling regions.

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: Utils/Settings.swift [hunk 1]
```diff
@@ -39,6 +39,6 @@ class Settings: NSObject {
         // Can't initialize a singleton
     }
     
-    // MARK: Shared Instance
+    // MARK:- Shared Instance
```
Comment (ceaglest) [inline Utils/Settings.swift]: Intentional?

File: AudioDeviceExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -260,6 +260,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
```
Comment (ceaglest) [inline AudioDeviceExample.xcodeproj/project.pbxproj]: I think there is iOS 9 specific code in this example. Can it be removed?

File: Utils/SettingsTableViewController.swift [hunk 3]
```diff
@@ -211,17 +211,7 @@ class SettingsTableViewController: UITableViewController {
...
-                if settings.videoCodec!.isKind(of: TVIVp8Codec.self) {
-                    if codec.isKind(of: TVIVp8Codec.self) {
-                        if ((codec as! TVIVp8Codec).isSimulcast == (settings.videoCodec as! TVIVp8Codec).isSimulcast) {
-                            selectedButton = selectionButton;
-                        }
-                    }
-                } else {
-                    selectedButton = selectionButton;
-                }
+                selectedButton = selectionButton;
```
Comment (ceaglest) [inline Utils/SettingsTableViewController.swift]: Nice find. 👍

File: ARKitExample/ViewController.swift [hunk 4]
```diff
@@ -103,11 +103,11 @@ class ViewController: UIViewController {
             return
         }
 
-        // As a TVIVideoSource, we must deliver CVPixelBuffers and not CGImages to the consumer.
+        // As a VideoSource, we must deliver CVPixelBuffers and not CGImages to the consumer.
```
Comment (ceaglest) [inline ARKitExample/ViewController.swift]: Not using first person "we":

// A VideoSource must deliver CVPixelBuffers (and not CGImages) to the VideoSink.

File: ARKitExample/ViewController.swift [hunk 5]
```diff
@@ -185,20 +185,20 @@ extension ViewController: ARSCNViewDelegate {
...
-    func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
+    func roomDidFailToConnect(room: Room, error: Error) {
         print("Failed to connect to a Room: \(error).")
 
-        let alertController = UIAlertController.init(title: "Connection Failed",
-                                                     message: "Couldn't connect to Room \(room.name). code:\(error._code) \(error.localizedDescription)",
-                                                     preferredStyle: UIAlertController.Style.alert)
+        let alertController = UIAlertController(title: "Connection Failed",
+                                                message: "Couldn't connect to Room \(room.name). code:\(error._code) \(error.localizedDescription)",
+            preferredStyle: UIAlertController.Style.alert)
```
Comment (ceaglest) [inline ARKitExample/ViewController.swift]: Indentation

File: AudioDeviceExample/ViewController.swift [hunk 6]
```diff
@@ -567,63 +551,49 @@ extension ViewController : TVIRemoteParticipantDelegate {
...
-                           error: Error,
-                           for participant: TVIRemoteParticipant) {
+    func didFailToSubscribeToAudioTrack(publication: RemoteAudioTrackPublication, error: Error, participant: RemoteParticipant) {
         logMessage(messageText: "FailedToSubscribe \(publication.trackName) audio track, error = \(String(describing: error))")
     }
 
-    func failedToSubscribe(toVideoTrack publication: TVIRemoteVideoTrackPublication,
-                           error: Error,
-                           for participant: TVIRemoteParticipant) {
+    func didFailToSubscribeToVideoTrack(publication: RemoteVideoTrackPublication, error: Error, participant: RemoteParticipant) {
```
Comment (ceaglest) [inline AudioDeviceExample/ViewController.swift]: Should we return labels when there are 3 or more args? Any hard and fast rule? :)

File: AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m [hunk 7]
```diff
@@ -2,7 +2,7 @@
 //  ExampleCoreAudioDevice.m
 //  AudioDeviceExample
 //
-//  Copyright © 2018 Twilio, Inc. All rights reserved.
+//  Copyright © 2018-2019 Twilio, Inc. All rights reserved.
```
Comment (ceaglest) [inline AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m]: This class has iOS 9 specific code for interruption handling which can now be removed.

File: README.md [hunk 8]
```diff
@@ -4,7 +4,7 @@
```
Comment (ceaglest) [inline README.md]: Update docs and download links to point directly at 3.0.0 / 3.0.0-beta1.

File: README.md [hunk 9]
```diff
@@ -80,12 +80,11 @@ You will also find additional examples that provide more advanced use cases of t
...
+- [ARKit](ARKitExample) - Captures augmented reality content with `ARKit` and uses `SceneKit` to place objects in the virtual scene. The resulting rendering is previewed locally, and shared in a `TVIRoom` via a custom `TVIVideoSource` source. This app requires Xcode 9 & iOS 11.
 - [CallKit](VideoCallKitQuickStart) - Shows how to use Twilio Video with the [CallKit](https://developer.apple.com/reference/callkit) framework.
 - [DataTrack](DataTrackExample) - Shows how to use the Data Track APIs for interactive drawing.
 - [MultiParty](MultiPartyExample) - Video conferencing in a Group Room with up to 4 Participants, dominant speaker detection and network quality events.
-- [ReplayKit](ReplayKitExample) - Shows how to use ReplayKit to share the screen, and microphone via `TVIVideoCapturer` and `TVIAudioDevice`.
-- [Screen Capturer](ScreenCapturerExample) - Use a custom `TVIVideoSource` to capture the contents of a `WKWebView`, and fallback to `TVIScreenCapturer` on older versions of iOS.
+- [ReplayKit](ReplayKitExample) - Shows how to use ReplayKit to share the screen, and microphone via `TVIVideoSource` and `TVIAudioDevice`.
```
Comment (ceaglest) [inline README.md]: Restore Screen Capturer entry, removing the mention of `TVIScreenCapturer`.

File: ARKitExample/ViewController.swift (hunk 4 above)
Comment (paynerc) [reply ARKitExample/ViewController.swift]: Done

File: ARKitExample/ViewController.swift (hunk 5 above)
Comment (paynerc) [reply ARKitExample/ViewController.swift]: Done

File: README.md (hunk 8 above)
Comment (paynerc) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (paynerc) [reply README.md]: Done. Well, as far as the `README.md` goes. 😉

File: AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m (hunk 7 above)
Comment (paynerc) [reply AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m]: Removed!

File: AudioDeviceExample.xcodeproj/project.pbxproj (hunk 2 above)
Comment (paynerc) [reply AudioDeviceExample.xcodeproj/project.pbxproj]: This has been removed.

**Pull request merged**